src/plugins/data/server/index_patterns/routes/runtime_fields/create_runtime_field.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { handleErrors } from '../util/handle_errors';
+import { runtimeFieldSpecSchema } from '../util/schemas';
+import { IRouter, StartServicesAccessor } from '../../../../../../core/server';
+import type { DataPluginStart, DataPluginStartDependencies } from '../../../plugin';
+
+export const registerCreateRuntimeFieldRoute = (
+  router: IRouter,
+  getStartServices: StartServicesAccessor<DataPluginStartDependencies, DataPluginStart>
+) => {
+  router.post(
+    {
+      path: '/api/index_patterns/index_pattern/{id}/runtime_field',
+      validate: {
+        params: schema.object({
+          id: schema.string({
+            minLength: 1,
+            maxLength: 1_000,
+          }),
+        }),
+        body: schema.object({
+          name: schema.string({
+            minLength: 1,
+            maxLength: 1_000,
+          }),
+          runtimeField: runtimeFieldSpecSchema,
+          // TODO: extend this API to support `custom label`, `count` and `format`?
```

Comment:
I don't like the idea of reusing the whole fieldSpec inside a runtime field API route. 
I think otherwise we should have implemented generic fields API and make it support runtime field. 
But we already went in a different direction with scripted fields route and I think this approach is fine

I suggest:

1. Keep runtime fields API minimal as is. I also like a minimal format of runtime field definition 
2. Use update fields API to update field formatters and other stuff. We can always extend runtime fields API later to support those, but for now, let's keep the API surface smaller

WDYT?

Reply:
I left runtime fields API as minimal as possible and added an integration test that attrs can be added using update fields endpoint